paasta_tools/graceful_app_drain.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+import argparse
+
+import service_configuration_lib
+from paasta_tools import marathon_tools
+from paasta_tools import drain_lib
+from paasta_tools import bounce_lib
+from paasta_tools.utils import decompose_job_id
+from paasta_tools.setup_marathon_job import do_bounce, get_old_live_draining_tasks
+from paasta_tools.utils import load_system_paasta_config
+from marathon.exceptions import MarathonHttpError
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='',
+    )
+    parser.add_argument(
+        'appname',
+        help="the app that will be drained",
+    )
+    parser.add_argument(
+        '-d', '--soa-dir',
+        dest="soa_dir",
+        metavar="SOA_DIR",
+        default=service_configuration_lib.DEFAULT_SOA_DIR,
+        help="define a different soa config directory",
+    )
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    full_appid = args.appname
```

Comment:
This particular script is designed to be used by admins in the situation where our bounce code can't proceed for whatever reason, and we need to gracefully kill a *specific app* to get it going. For this reason, the appid should not be calculated, it needs to be provided by an admin. This script is not meant to be used very often, and not by normal users.